### PR TITLE
feat: add conditions to unpacker

### PR DESCRIPTION
**Pull request by Skarlso**
This update adds conditions to unpacker including a `shouldReconcile` basic function that deals with last applied component verison.

### CODE REVIEW

File: controllers/unpacker_controller.go [hunk 1]
```diff
@@ -128,54 +129,139 @@ func (r *UnpackerReconciler) SetupWithManager(mgr ctrl.Manager) error {
...
-	logger := log.FromContext(ctx)
-
-	cv, err := r.getComponentVersion(ctx, obj.Spec.ComponentVersionRef.Namespace, obj.Spec.ComponentVersionRef.Name)
-	if err != nil {
-		return ctrl.Result{RequeueAfter: obj.GetRequeueAfter()}, err
+func (r *UnpackerReconciler) shouldReconcile(cv *v1alpha1.ComponentVersion, obj *deliveryv1alpha1.Unpacker) (bool, error) {
+	// If there is a mismatch between the observed generation of a component version, we trigger
+	// a reconcile. There is either a new version available or a dependent component version
+	// finished its reconcile process.
+	if obj.Status.LastAppliedComponentVersion != cv.Status.ReconciledVersion {
```
Comment (Skarlso) [inline controllers/unpacker_controller.go:250]: @phoban01 I was wondering if I should add piplinetemplate's source in here... 🤔

**Comment by Skarlso** [comment]
Screenshot 2023-02-14 at 17 28 47


Looks okay.

File: controllers/unpacker_controller.go (hunk 1 above)
Comment (phoban01) [reply controllers/unpacker_controller.go:250]: Yes definitely... anything that could result in a change to the output.

Comment (Skarlso) [reply controllers/unpacker_controller.go:250]: Okay, cool, thanks. I'll add that.

**Review by phoban01 — COMMENTED** [review]
We should add some more description conditions or at least reasons to the api. These could help provide information about why the pipeline evaluation failed or so on.

**Comment by Skarlso** [comment]
>We should add some more description conditions or at least reasons to the api. These could help provide information about why the pipeline evaluation failed or so on.

I thought the reason is the error if there were any, otherwise, it's either success or still going, not?

**Comment by phoban01** [comment]
The Reason gives higher level context, the message would contain the actual specifics of the error.

**Comment by Skarlso** [comment]
@phoban01 I added a bunch of failure reasons. I think this is what you meant? Sorry, I think I missed that previously. ALSO... This made me realise that ocm-controller is missing a few failure reasons. :)

**Review by phoban01 — APPROVED** [review]
Great, thanks @Skarlso !

**Pull request merged**